### PR TITLE
Remove unused and deprecated things from kernel config

### PR DESCRIPTION
**Pull request by benbc**

### CODE REVIEW

**Comment by chrisvest** [comment]
Nice.

**Comment by benbc** [comment]
@jakewins Please could you review this PR for me?

File: community/kernel/src/main/java/org/neo4j/helpers/Settings.java [hunk 1]
```diff
@@ -1,943 +0,0 @@
...
- * A parser is a function that takes a string and converts to some type T. The parser may throw IllegalArgumentException
- * if it fails.
- *
- * The default value is the string representation of what you want as default. Special values are the constants NO_DEFAULT,
- * which means that you don't want any default value at all, and MANDATORY, which means that the user has to specify a value
- * for this setting. Not providing a mandatory value for a setting leads to an IllegalArgumentException.
- *
- * @deprecated this class is deprecated and will be moved to internal packages in the next major release
- */
-@Deprecated
```
Comment (jakewins) [inline community/kernel/src/main/java/org/neo4j/helpers/Settings.java:55]: Unfortunately, this was not deprecated in 2.x :( https://github.com/neo4j/neo4j/commit/1ba7202f896b04d53bb14a68972dca170895617c

**Comment by jakewins** [comment]
Other than the comment above, this looks absolutely brilliant to me.

File: community/kernel/src/main/java/org/neo4j/helpers/Settings.java (hunk 1 above)
Comment (benbc) [inline community/kernel/src/main/java/org/neo4j/helpers/Settings.java:55]: But this package part of the public API, is it?

Comment (jakewins) [inline community/kernel/src/main/java/org/neo4j/helpers/Settings.java:55]: Oh, true, and it wasn't there back when org.neo4j.kernel was exposed as public API. Coolio.

**Pull request merged**